### PR TITLE
Testing whether a point is inside a shape

### PR DESCRIPTION
**Pull request by jason-bowen-zheng**
Some functions were added to several shapes:

1. Circle
2. Ellipse
3. Sector
4. Line
5. Rectangle
6. BoderedRectangle
7. Triangle
8. Star
9. Polygon

The following code
```python
from pyglet import shapes

rect = Rectangle(...)
# Do something else
if (100, 100) in rect:
    rect.color = (255, 0, 0)
```
will turn `rect`'s color to red when the point `(100, 100)` is in the shape.

It's also available for rotated and anchored shapes.

### CODE REVIEW

File: pyglet/shapes.py [hunk 1]
```diff
@@ -719,6 +731,10 @@ def __init__(self, x, y, radius, segments=None, color=(255, 255, 255, 255),
         self._create_vertex_list()
         self._update_vertices()
 
+    def __contains__(self, point):
+        assert len(point) == 2
+        return math.dist((self._x - self._anchor_x, self._y - self._anchor_y), point) < self._radius
```
Comment (gran4) [inline pyglet/shapes.py:736]: 1: You have __contains__ twice
2: You don't actually check if the point is in any of the shapes

Comment (jason-bowen-zheng) [reply pyglet/shapes.py:736]: Can you explain it in detail?

**Comment by benmoran56** [comment]
@jason-bowen-zheng 
I can see you're still working on this. Please let me know once you feel ready.

File: pyglet/shapes.py (hunk 1 above)
Comment (gran4) [reply pyglet/shapes.py:751]: I am dumb never mind

**Comment by jason-bowen-zheng** [comment]
@benmoran56, I think it's nearly finished. But there are still some problems:

1. only regular polygons were tested
2. the SAT implementation may have problem: [triangle's](https://github.com/pyglet/pyglet/blob/41f4f40a375576fa82240da50a64549a9f431c33/pyglet/shapes.py#L1554) and [polygon's](https://github.com/pyglet/pyglet/blob/41f4f40a375576fa82240da50a64549a9f431c33/pyglet/shapes.py#L1791) codes that gets the correct result is different.

**Comment by benmoran56** [comment]
I haven't checked the *sat* function yet on my own PC, but could it be an issue with convex v.s. concave polygons?

Perhaps a different algorithm might be better, that can handle both types of polygons. This one might be good to try: https://www.algorithms-and-technologies.com/point_in_polygon/python

**Comment by jason-bowen-zheng** [comment]
First, SAT only supports convex polygons(since pyglet only support convex polygons, too).

Then, `point_in_polygon` still has some problems.

So my opinion is that we can merge this PR first. Then make pyglet support concave polygons and `point_in_polygon`.

**Comment by benmoran56** [comment]
Perhaps it's enough to treat the Star as a circle :thinking:

**Comment by jason-bowen-zheng** [comment]
This is indeed a compromise. The radius of a circle can be considered as `(inner_radius+outter_radius)/2`.

**Comment by benmoran56** [comment]
Great! 
That's all of the shapes then, isn't it?

**Comment by jason-bowen-zheng** [comment]
Yes, except for those drawn using `GL_LINES`.

**Review by benmoran56 — APPROVED** [review]

**Comment by jason-bowen-zheng** [comment]
I think now this PR is perfect!

Besides, I made the docs of shapes short and brief.

**Review by benmoran56 — APPROVED** [review]
Looks good!

**Pull request merged**